nav2_docking/opennav_docking/include/opennav_docking/simple_charging_dock.hpp

```diff
@@ -149,6 +161,28 @@ class SimpleChargingDock : public opennav_docking_core::ChargingDock
 
   rclcpp_lifecycle::LifecycleNode::SharedPtr node_;
   std::shared_ptr<tf2_ros::Buffer> tf2_buffer_;
+
+  // Detector control parameters
+  std::string detector_service_name_;
+  double detector_service_timeout_{5.0};
+  bool subscribe_toggle_{false};
+
+  // Client used to call the Trigger service
+  nav2_util::ServiceClient<
+    std_srvs::srv::Trigger,
+    rclcpp_lifecycle::LifecycleNode::SharedPtr>::SharedPtr detector_client_;
+
+  // Dynamic subscription pointer (can be reset to release CPU/GPU)
+  rclcpp::Subscription<geometry_msgs::msg::PoseStamped>::SharedPtr detected_pose_sub_;
```

Comment:
Ditto, see https://docs.nav2.org/migration/Kilted.html#new-nav2-ros-common-nav2-lifecycle-node needs to be a nav2::Subscription